phpunit.xml.dist

```diff
@@ -2,30 +2,34 @@
 
 <!-- http://phpunit.de/manual/4.1/en/appendixes.configuration.html -->
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```

Comment:
Are the changes in this file mandatory? If we can keep it "as-is", we ensure that the project still work as expected with older versions.

Reply:
they are necessary.  I have converted our own testsuite to be PHPUnit 10 compatible (there are BC breaks).

I don't think there is any problem in our project requiring a dev dependency explicitly of phpunit 10, as our tests have been updated and now require 10.  since it is a dev dependency and not a hard dependency, consumers can pull in Panther and use whatever PHPUnit they provide in their project's requirements.

Reply:
It's an optional dependency, but not only a dev one. We must ensure in our CI that Panther still run on projects using PHPUnit <10.

Reply:
I'm not sure how you want to proceed here.  In order to make Panther's internal tests phpunit 10 compliant, I had to make various changes you'll see to the test files.  additionally, I had to migrate to `phpunit.xml.dist` structure/formatting/config.

I am not immediately certain how we can make our own internal test suite runnable on 10 and >10. It looks like the `doctrine-test-bundle` created two different `phpunit.xml` files, one for <10 and one for 10, so maybe there is a solution somehow: https://github.com/dmaicher/doctrine-test-bundle/tree/master/tests

In short, I'm not sure what you are asking here... do you want me to revert the changes to `phpunit.xml.dist` and then roll back all the PHPUnit 10 specific changes to our internal test files (`DataProvider` attribute, etc)? Or so I need to figure out how our internal testsuite can pass both PHPUnit8/9 and also PHPUnit10?

Reply:
If it's not possible to use a PHPUnit configuration that is compatible with both PHPUnit 10 and previous versions, then yes having two files will be necessary.

Reply:
The two files thing isn't really the problem, that's the easy part.  The potential problem is having internal tests that are compatible with 10 and <10.  But doctrine-test-bundle seems to be doing it, so I will give it a go and report back.

Reply:
Before wasting your time, I would like to hear what's @nicolas-grekas opinion on this.

Do we need to do so much efforts to support old PHPUnit versions, or can we expect a PHPUnit Bridge version compatible with PHPUnit soon and just release a new major?

Reply:
actually, may not be worth wasting _his_ time.  I spoke before testing.  I think I have this working. Commit incoming...

Reply:
done: https://github.com/arderyp/panther/commit/3c0c3f6f7d300bacdd4c1ce5fd42a10018edf07f

there is a new `phpunit.xml.dist.10` file for usage in PHPUnit 10 GitHub Action

@dunglas please resolve if sufficient